third_party/libunwindstack/PeCoffUnwindInfoUnwinderX86_64.cpp

```diff
@@ -0,0 +1,240 @@
+                                          const UnwindInfo& unwind_info,
+                                          uint64_t current_code_offset, uint64_t* frame_pointer,
+                                          bool* frame_pointer_used) {
+  // Data is parsed from the object file, so we have to assume that it may be inconsistent.
+  if (unwind_info.num_codes != unwind_info.unwind_codes.size()) {
+    last_error_.code = ERROR_INVALID_COFF;
+    return false;
+  }
+
+  RegsImpl<uint64_t>* cur_regs = reinterpret_cast<RegsImpl<uint64_t>*>(regs);
+
+  for (int op_idx = 0; op_idx < unwind_info.num_codes;) {
+    UnwindCode unwind_code = unwind_info.unwind_codes[op_idx];
+    switch (unwind_code.GetUnwindOp()) {
+      case UWOP_PUSH_NONVOL: {
+        if (unwind_code.code_and_op.code_offset > current_code_offset) {
+          op_idx += 1;
+          continue;
+        }
+        uint64_t register_value;
+        if (!process_memory->Read64(cur_regs->sp(), &register_value)) {
+          last_error_.code = ERROR_MEMORY_INVALID;
+          last_error_.address = cur_regs->sp();
+          return false;
+        }
+        cur_regs->set_sp(cur_regs->sp() + sizeof(uint64_t));
+
+        uint8_t op_info = unwind_code.GetOpInfo();
+        uint16_t reg = MapToUnwindstackRegister(op_info);
+        (*cur_regs)[reg] = register_value;
+
+        op_idx++;
+        break;
+      }
+      case UWOP_ALLOC_LARGE: {
+        uint8_t op_info = unwind_code.GetOpInfo();
+        uint32_t allocation_size = 0;
+
+        if (op_info == 0) {
+          if (unwind_code.code_and_op.code_offset > current_code_offset) {
+            // Must be the total number of indices we have to increase by.
+            op_idx += 2;
+            continue;
+          }
+          if (op_idx + 1 >= unwind_info.num_codes) {
+            last_error_.code = ERROR_INVALID_COFF;
+            return false;
+          }
+          UnwindCode offset = unwind_info.unwind_codes[op_idx + 1];
+          allocation_size = 8 * static_cast<uint32_t>(offset.frame_offset);
+          op_idx += 1;
+        } else if (op_info == 1) {
+          if (unwind_code.code_and_op.code_offset > current_code_offset) {
+            // Must be the total number of indices we have to increase by.
+            op_idx += 3;
+            continue;
+          }
+          if (op_idx + 2 >= unwind_info.num_codes) {
+            last_error_.code = ERROR_INVALID_COFF;
+            return false;
+          }
+          UnwindCode offset1 = unwind_info.unwind_codes[op_idx + 1];
+          UnwindCode offset2 = unwind_info.unwind_codes[op_idx + 2];
+
+          allocation_size = static_cast<uint32_t>(offset1.frame_offset) +
+                            (static_cast<uint32_t>(offset2.frame_offset) << 16);
+          op_idx += 2;
+        } else {
+          last_error_.code = ERROR_INVALID_COFF;
+          return false;
+        }
+
+        cur_regs->set_sp(cur_regs->sp() + allocation_size);
+        op_idx += 1;
```

Comment:
NIt: Splitting up the increment of `op_idx` into two steps here makes it a little more confusing to me? I personally don't see a downside in incrementing one more in both if-branches - especially since op_idx is not used here.

Reply:
Yes, agree, in particular since the lines before the `continue` above increase by the larger number.